staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"sync/atomic"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
```

Comment:
there are edges around handling of paths containing symlinks we want to make sure we handle (since that's how a mounted-in configmap would surface). I *think* we have to watch parent dirs as well in case they are symlinks

Reply:
Added tests to handle EC file updated by
- truncate (os.WriteFile)
- deleteAndCreate (os.Remove and then os.Create)
- move (write new config in another tmp dir then os.Rename to move it to dir being watched by watchers)
- symLink (created config in parentDir, created symLink of same in dir being watched by watchers, and then update parent dir.)